### PR TITLE
Add Thanos sidecar metrics + alerts

### PR DESCRIPTION
**Pull request by simonpasquier**
* [X] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by simonpasquier** [comment]
This adds `kube-rbac-proxy` in front of the Thanos sidecar. The proxies are configured to allow only the `/metrics` path.

There are 2 "versions" of the change:
* e93bfc0 adds the Thanos sidecar port to the existing Prometheus services. It doesn't require the creation of a new service but diverges from what kube-prometheus does.
* 2c7d548 follows the kube-prometheus way with new services for the Thanos sidecars.

**Comment by paulfantom** [comment]
This also fixes [BZ1878766](https://bugzilla.redhat.com/show_bug.cgi?id=1878766)

File: jsonnet/prometheus-user-workload.jsonnet [hunk 1]
```diff
@@ -259,6 +283,39 @@ local sccRole =
                 },
               ],
             },
+            {
+              name: 'kube-rbac-proxy-thanos',
+              image: $._config.imageRepos.kubeRbacProxy + ':' + $._config.versions.kubeRbacProxy,
```
Comment (lilic) [inline jsonnet/prometheus-user-workload.jsonnet:288]: We should bump this to https://github.com/brancz/kube-rbac-proxy/releases/tag/v0.8.0

File: test/e2e/prometheus_test.go [hunk 2]
```diff
@@ -27,6 +28,34 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func TestPrometheusMetrics(t *testing.T) {
```
Comment (lilic) [inline test/e2e/prometheus_test.go:31]: Nice!

**Review by lilic — COMMENTED** [review]
Nice work!

**Comment by simonpasquier** [comment]
/hold
waiting for #985

**Comment by paulfantom** [comment]
/unhold

#985 is merged

**Review by lilic — COMMENTED** [review]
/lgtm

**Pull request merged**